diff --git a/Utilities/cmlibuv/CMakeLists.txt b/Utilities/cmlibuv/CMakeLists.txt
index a0b161b..2144378 100644
--- a/Utilities/cmlibuv/CMakeLists.txt
+++ b/Utilities/cmlibuv/CMakeLists.txt
@@ -164,8 +164,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "CYGWIN" OR CMAKE_SYSTEM_NAME MATCHES "MSYS")
   list(APPEND uv_headers
     include/uv/posix.h
     )
-  list(APPEND uv_defines
-    )
+  list(APPEND uv_defines _GNU_SOURCE)
   list(APPEND uv_sources
     src/unix/cygwin.c
     src/unix/bsd-ifaddrs.c
