commit ace1a2576036a76999684de2d2da7b4d4e87ed32
Author: David McFarland <corngood@gmail.com>
Date:   Sat Nov 25 20:31:56 2017 -0400

    p2

diff --git a/lib/Munin/Common/Defaults.pm.PL b/lib/Munin/Common/Defaults.pm.PL
index 13eb5fdf..6a16d391 100644
--- a/lib/Munin/Common/Defaults.pm.PL
+++ b/lib/Munin/Common/Defaults.pm.PL
@@ -92,7 +92,7 @@ sub export_to_environment {
 
     my %defaults = %{ $class->get_defaults() };
     while ( my ( $k, $v ) = each %defaults ) {
-        $ENV{$k} = $v;
+        $ENV{$k} = $ENV{$k} || $v;
     }
 
     return;
diff --git a/lib/Munin/Node/Service.pm b/lib/Munin/Node/Service.pm
index fbb773ee..eecc22d7 100644
--- a/lib/Munin/Node/Service.pm
+++ b/lib/Munin/Node/Service.pm
@@ -121,7 +121,7 @@ sub export_service_environment {
     # We append the USER to the MUNIN_PLUGSTATE, to avoid CVE-2012-3512
     my $uid = $self->_resolve_uid($service);
     my $user = getpwuid($uid);
-    $ENV{MUNIN_PLUGSTATE} = "$Munin::Common::Defaults::MUNIN_PLUGSTATE/$user";
+    $ENV{MUNIN_PLUGSTATE} = "$ENV{MUNIN_PLUGSTATE}/$user";
 
     # Provide a consistent default state-file.
     $ENV{MUNIN_STATEFILE} = "$ENV{MUNIN_PLUGSTATE}/$service-$ENV{MUNIN_MASTER_IP}";
@@ -243,7 +243,7 @@ sub exec_service
 
     # XXX - Create the statedir for the user
     my $uid = $self->_resolve_uid($service);
-    Munin::Node::OS->mkdir_subdir("$Munin::Common::Defaults::MUNIN_PLUGSTATE", $uid);
+    Munin::Node::OS->mkdir_subdir("$ENV{MUNIN_PLUGSTATE}", $uid);
 
     $self->change_real_and_effective_user_and_group($service);
 
