From faef644be402b4cc0e04b644b99663f29a537241 Mon Sep 17 00:00:00 2001
From: David McFarland <corngood@gmail.com>
Date: Sun, 9 Mar 2025 15:54:09 -0300
Subject: [PATCH] SConstruct: inherit external environment

---
 SConstruct | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SConstruct b/SConstruct
index f4500d7525..c3f1702beb 100644
--- a/SConstruct
+++ b/SConstruct
@@ -112,6 +112,7 @@ for x in sorted(glob.glob("platform/*")):
 # further down after parsing our platform-specific configuration.
 # Then we prepend PATH to make it take precedence, while preserving SCons' own entries.
 env = Environment(tools=[])
+env["ENV"] = os.environ
 env.PrependENVPath("PATH", os.getenv("PATH"))
 env.PrependENVPath("PKG_CONFIG_PATH", os.getenv("PKG_CONFIG_PATH"))
 if "TERM" in os.environ:  # Used for colored output.
-- 
2.47.1

