From 935f101d5709d2f9e0591e3dc27f643b4f4b7e88 Mon Sep 17 00:00:00 2001
From: David McFarland <corngood@gmail.com>
Date: Sun, 9 Mar 2025 19:55:56 -0300
Subject: [PATCH] SConstruct: disable builtin_* by default

---
 SConstruct | 58 ++++++++++++++++++++++++++++--------------------------
 1 file changed, 30 insertions(+), 28 deletions(-)

diff --git a/SConstruct b/SConstruct
index c3f1702beb..b078531908 100644
--- a/SConstruct
+++ b/SConstruct
@@ -239,34 +239,36 @@ opts.Add("cache_path", "Path to a directory where SCons cache files will be stor
 opts.Add("cache_limit", "Max size (in GiB) for the SCons cache. 0 means no limit.", "0")
 
 # Thirdparty libraries
-opts.Add(BoolVariable("builtin_brotli", "Use the built-in Brotli library", True))
-opts.Add(BoolVariable("builtin_certs", "Use the built-in SSL certificates bundles", True))
-opts.Add(BoolVariable("builtin_clipper2", "Use the built-in Clipper2 library", True))
-opts.Add(BoolVariable("builtin_embree", "Use the built-in Embree library", True))
-opts.Add(BoolVariable("builtin_enet", "Use the built-in ENet library", True))
-opts.Add(BoolVariable("builtin_freetype", "Use the built-in FreeType library", True))
-opts.Add(BoolVariable("builtin_msdfgen", "Use the built-in MSDFgen library", True))
-opts.Add(BoolVariable("builtin_glslang", "Use the built-in glslang library", True))
-opts.Add(BoolVariable("builtin_graphite", "Use the built-in Graphite library", True))
-opts.Add(BoolVariable("builtin_harfbuzz", "Use the built-in HarfBuzz library", True))
-opts.Add(BoolVariable("builtin_icu4c", "Use the built-in ICU library", True))
-opts.Add(BoolVariable("builtin_libogg", "Use the built-in libogg library", True))
-opts.Add(BoolVariable("builtin_libpng", "Use the built-in libpng library", True))
-opts.Add(BoolVariable("builtin_libtheora", "Use the built-in libtheora library", True))
-opts.Add(BoolVariable("builtin_libvorbis", "Use the built-in libvorbis library", True))
-opts.Add(BoolVariable("builtin_libwebp", "Use the built-in libwebp library", True))
-opts.Add(BoolVariable("builtin_wslay", "Use the built-in wslay library", True))
-opts.Add(BoolVariable("builtin_mbedtls", "Use the built-in mbedTLS library", True))
-opts.Add(BoolVariable("builtin_miniupnpc", "Use the built-in miniupnpc library", True))
-opts.Add(BoolVariable("builtin_openxr", "Use the built-in OpenXR library", True))
-opts.Add(BoolVariable("builtin_pcre2", "Use the built-in PCRE2 library", True))
-opts.Add(BoolVariable("builtin_pcre2_with_jit", "Use JIT compiler for the built-in PCRE2 library", True))
-opts.Add(BoolVariable("builtin_recastnavigation", "Use the built-in Recast navigation library", True))
-opts.Add(BoolVariable("builtin_rvo2_2d", "Use the built-in RVO2 2D library", True))
-opts.Add(BoolVariable("builtin_rvo2_3d", "Use the built-in RVO2 3D library", True))
-opts.Add(BoolVariable("builtin_xatlas", "Use the built-in xatlas library", True))
-opts.Add(BoolVariable("builtin_zlib", "Use the built-in zlib library", True))
-opts.Add(BoolVariable("builtin_zstd", "Use the built-in Zstd library", True))
+# START
+opts.Add(BoolVariable("builtin_brotli", "Use the built-in Brotli library", False))
+opts.Add(BoolVariable("builtin_certs", "Use the built-in SSL certificates bundles", False))
+opts.Add(BoolVariable("builtin_clipper2", "Use the built-in Clipper2 library", False))
+opts.Add(BoolVariable("builtin_embree", "Use the built-in Embree library", False))
+opts.Add(BoolVariable("builtin_enet", "Use the built-in ENet library", False))
+opts.Add(BoolVariable("builtin_freetype", "Use the built-in FreeType library", False))
+opts.Add(BoolVariable("builtin_msdfgen", "Use the built-in MSDFgen library", False))
+opts.Add(BoolVariable("builtin_glslang", "Use the built-in glslang library", False))
+opts.Add(BoolVariable("builtin_graphite", "Use the built-in Graphite library", False))
+opts.Add(BoolVariable("builtin_harfbuzz", "Use the built-in HarfBuzz library", False))
+opts.Add(BoolVariable("builtin_icu4c", "Use the built-in ICU library", False))
+opts.Add(BoolVariable("builtin_libogg", "Use the built-in libogg library", False))
+opts.Add(BoolVariable("builtin_libpng", "Use the built-in libpng library", False))
+opts.Add(BoolVariable("builtin_libtheora", "Use the built-in libtheora library", False))
+opts.Add(BoolVariable("builtin_libvorbis", "Use the built-in libvorbis library", False))
+opts.Add(BoolVariable("builtin_libwebp", "Use the built-in libwebp library", False))
+opts.Add(BoolVariable("builtin_wslay", "Use the built-in wslay library", False))
+opts.Add(BoolVariable("builtin_mbedtls", "Use the built-in mbedTLS library", False))
+opts.Add(BoolVariable("builtin_miniupnpc", "Use the built-in miniupnpc library", False))
+opts.Add(BoolVariable("builtin_openxr", "Use the built-in OpenXR library", False))
+opts.Add(BoolVariable("builtin_pcre2", "Use the built-in PCRE2 library", False))
+opts.Add(BoolVariable("builtin_pcre2_with_jit", "Use JIT compiler for the built-in PCRE2 library", False))
+opts.Add(BoolVariable("builtin_recastnavigation", "Use the built-in Recast navigation library", False))
+opts.Add(BoolVariable("builtin_rvo2_2d", "Use the built-in RVO2 2D library", False))
+opts.Add(BoolVariable("builtin_rvo2_3d", "Use the built-in RVO2 3D library", False))
+opts.Add(BoolVariable("builtin_xatlas", "Use the built-in xatlas library", False))
+opts.Add(BoolVariable("builtin_zlib", "Use the built-in zlib library", False))
+opts.Add(BoolVariable("builtin_zstd", "Use the built-in Zstd library", False))
+# END
 
 # Compilation environment setup
 # CXX, CC, and LINK directly set the equivalent `env` values (which may still
-- 
2.47.1

