diff --git a/Makefile b/Makefile
index 9be557f..76d6b01 100644
--- a/Makefile
+++ b/Makefile
@@ -158,7 +158,7 @@ REGRESS_ARGS	+= "--parse-no-deflists"
 REGRESS_ENV	 = LC_ALL=en_US.UTF-8
 
 all: bins lowdown.pc liblowdown.so
-bins: lowdown lowdown-diff
+bins: lowdown.exe lowdown-diff
 
 www: all $(HTMLS) $(PDFS) $(THUMBS) lowdown.tar.gz lowdown.tar.gz.sha512
 
@@ -170,11 +170,11 @@ installwww: www
 	$(INSTALL) -m 0444 lowdown.tar.gz $(WWWDIR)/snapshots
 	$(INSTALL) -m 0444 lowdown.tar.gz.sha512 $(WWWDIR)/snapshots
 
-lowdown: liblowdown.a main.o
+lowdown.exe: liblowdown.a main.o
 	$(CC) -o $@ main.o liblowdown.a $(LDFLAGS) $(LDADD_MD5) -lm $(LDADD)
 
-lowdown-diff: lowdown
-	ln -f lowdown lowdown-diff
+lowdown-diff: lowdown.exe
+	ln -f lowdown.exe lowdown-diff
 
 liblowdown.a: $(OBJS) $(COMPAT_OBJS)
 	$(AR) rs $@ $(OBJS) $(COMPAT_OBJS)
@@ -206,7 +206,7 @@ install: bins
 	$(INSTALL_DATA) share/man/* $(DESTDIR)$(SHAREDIR)/lowdown/man
 	$(INSTALL_DATA) share/ms/* $(DESTDIR)$(SHAREDIR)/lowdown/ms
 	$(INSTALL_DATA) share/odt/* $(DESTDIR)$(SHAREDIR)/lowdown/odt
-	$(INSTALL_PROGRAM) lowdown $(DESTDIR)$(BINDIR)
+	$(INSTALL_PROGRAM) lowdown.exe $(DESTDIR)$(BINDIR)
 	$(INSTALL_PROGRAM) lowdown-diff $(DESTDIR)$(BINDIR)
 	for f in $(MAN1S) $(MAN5S) ; do \
 		name=`basename $$f .html` ; \
