From 366f2a6573685e3d6b4007bbd05ab94e9c87ac14 Mon Sep 17 00:00:00 2001
From: David McFarland <corngood@gmail.com>
Date: Fri, 31 Aug 2018 09:11:18 -0300
Subject: [PATCH] fix m4 glib_DEFUN redefinition on cygwin

---
 glib/m4macros/glib-gettext.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/m4macros/glib-gettext.m4 b/glib/m4macros/glib-gettext.m4
index 5217fd8..9f39b5f 100644
--- a/glib/m4macros/glib-gettext.m4
+++ b/glib/m4macros/glib-gettext.m4
@@ -36,8 +36,8 @@ dnl We go to great lengths to make sure that aclocal won't
 dnl try to pull in the installed version of these macros
 dnl when running aclocal in the glib directory.
 dnl
-m4_copy([AC_DEFUN],[glib_DEFUN])
-m4_copy([AC_REQUIRE],[glib_REQUIRE])
+m4_copy_force([AC_DEFUN],[glib_DEFUN])
+m4_copy_force([AC_REQUIRE],[glib_REQUIRE])
 dnl
 dnl At the end, if we're not within glib, we'll define the public
 dnl definitions in terms of our private definitions.
-- 
2.17.1.windows.2.3.ge2d2aac684.dirty

